### DISCUSSION TITLE
Python syntax highlighting in a YAML document

### DISCUSSION THREAD

**Discussion by hjakasa** [microsoft/monaco-editor#4034] [Q&A]
I have a use case where users will be writing python code within a YAML document. There will be a tag (!Python) after which all text will be treated as python code. I've gotten python autocompletion to work based on a list of keywords and some regex with a CompletionItemProvider, but not seeing a path forward for syntax highlighting.

I have an editor instance which has setModelLanguage which I can use to switch between python and yaml syntax highlighting for the entire editor (which I'm doing currently as a work around) but ideally there is a simple way to apply text highlighting for certain line numbers in the editor. Maybe there's a way to do this through tokenizers?

**Comment by ylmzfurkan** [comment]
Hii @hjakasa 

You can use a library like Pygments to highlight the Python code. Pygments is a Python library that can be used to highlight code in a variety of languages, including Python. You can use Pygments to highlight the Python code in your YAML document and then display the highlighted code in your text editor.
You can use a web service like Trinket or Codepen to highlight the Python code in your YAML document. Trinket and Codepen are web services that allow you to write and run code in a variety of languages, including Python. You can use Trinket or Codepen to highlight the Python code in your YAML document and then share the highlighted code with others.
You can use a plugin for your text editor to highlight the Python code in your YAML document. There are a number of plugins available for text editors like Sublime Text, Atom, and Visual Studio Code that can be used to highlight the Python code in your YAML document.
Here is an example of how you would use Pygments to highlight the Python code in a YAML document:

```
import pygments
import pygments.lexers
import pygments.formatters

def highlight_python(text):
    lexer = pygments.lexers.PythonLexer()
    formatter = pygments.formatters.TerminalFormatter()
    code = pygments.highlight(text, lexer, formatter)
    return code

if __name__ == '__main__':
    text = '`python\nprint("Hello, world!")\n`'
    print(highlight_python(text))
```

This code will highlight the Python code in the text variable and then print the highlighted code to the console.

**Reply by hjakasa** [reply]
How would this work if I'm using the monaco editor type script package? I have an editor instance which has setModelLanguage which I can use to switch between python and yaml syntax highlighting for the entire editor (which I'm doing currently as a work around) but ideally there is a simple way to apply text highlighting for certain line numbers in the editor. Maybe there's a way to do this through tokenizers?

**Reply by hjakasa** [reply]
Oh, just realized this is a gpt answer :)
